components/clarity-lsp/src/common/requests/definitions.rs

```diff
@@ -470,10 +473,16 @@ impl<'a> ASTVisitor<'a> for Definitions {
     ) -> bool {
         let local_scope = || -> Option<HashMap<ClarityName, Range>> {
             let mut result = HashMap::new();
+
             let binding_exprs = expr.match_list()?.get(1)?.match_list()?;
             for binding in binding_exprs {
-                let (name, _) = binding.match_list()?.split_first()?;
-                result.insert(name.match_atom()?.to_owned(), span_to_range(&binding.span));
+                if let Some(name) = binding
+                    .match_list()
+                    .and_then(|l| l.split_first())
+                    .and_then(|(name, _)| name.match_atom())
+                {
+                    result.insert(name.to_owned(), span_to_range(&binding.span));
+                }
```

Comment:
This new approach is nicer to read, but not equivalent to the previous implementation. In the initial approach, the `?` would break the loop on the 1st occurence, with your refactoring, we would keep looping.
Is that what we want?

Reply:
Yes that intended. There's the same logic in a couple of other places in this PR.
I might have been over using `?` in methods returns `Option<>`, _especially_ in loops.
With the `.and_then`s, the code is a bit more resilient to invalid ASTs (which happens a lot while the contract is being edited). So if an expression is invalid, it's just ignore and we continue with the loops 👌